File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        answer = 0

        for i, ch in enumerate(s):
            position = i + 1

            normal_position = ord(ch) - ord('a') + 1

            reverse_position = 27 - normal_position

            answer += reverse_position * position

        return answer

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {

        int answer = 0;

        for (int i = 0; i < s.length(); i++) {

            char ch = s.charAt(i);

            int position = i + 1;

            int normalPosition = ch - 'a' + 1;

            int reversePosition = 27 - normalPosition;

            answer += reversePosition * position;
        }

        return answer;
    }
}